### DISCUSSION TITLE
dotnet lambda deploy-function AspNetCoreWebAPI does not infer runtime from project file

### DISCUSSION THREAD

**Discussion by bjhogan** [aws/aws-lambda-dotnet#1199] [Q&A] [answered]
### Describe the bug

Create a new serverless Lambda function - 

`dotnet new serverless.AspNetCoreWebAPI -n AspNetCoreWebAPI`

Change to the `AspNetCoreWebAPI/src/AspNetCoreWebAPI`

Attempt to deploy the function -

```
dotnet lambda deploy-function AspNetCoreWebAPI       
Amazon Lambda Tools for .NET Core applications (5.4.2)
Project Home: https://github.com/aws/aws-extensions-for-dotnet-cli, https://github.com/aws/aws-lambda-dotnet

Enter Runtime: (The runtime environment for the Lambda function)
```

Runtime should not be requested. 

### Expected Behavior

I expect the tooling to infer the runtime from the csproj file, or that the runtime in included in the `aws-lambda-tools-defaults.json` file. 

### Current Behavior

User is asked to enter the runtime.

### Reproduction Steps

See description.

### AWS .NET SDK and/or Package version used

Amazon Lambda Tools for .NET Core applications (5.4.2)

### Targeted .NET Platform

.NET 6

### Operating System and version

Windows 10

**Comment by ashishdhingra** [comment, marked as answer]
@bjhogan Thanks for reporting the issue. I'm curious that you are using `serverless.AspNetCoreWebAPI` template, but using the `dotnet lambda deploy-function` command. Shouldn't you be using `dotnet lambda deploy-serverless` command? Please check.

**Comment by bjhogan** [comment]
Hi @ashishdhingra, you are right! Been working on too many `lambda.*` templates lately and forgot the `deploy-serverless`. 

Maybe the ticket should be - "Don't allow `dotnet lambda deploy-function...` to run if there is `serverless.template` present? 
Would be a better user experience?

**Reply by ashishdhingra** [reply]
> Hi @ashishdhingra, you are right! Been working on too many `lambda.*` templates lately and forgot the `deploy-serverless`.
> 
> Maybe the ticket should be - "Don't allow `dotnet lambda deploy-function...` to run if there is `serverless.template` present? Would be a better user experience?

@bjhogan It might not a good idea to add proposed validation. The respective subcommands are in place to be used with respective project types. What if we have the project for simple lambda function, but somehow user also includes `serverless.template` in the project accidentally (rare case, but possible)? This would add unnecessary complexity.

**Reply by bjhogan** [reply]
@ashishdhingra  How about having only the `serverless.template` file in serverless project templates, do not include the `aws-lambda-tools-defaults.json`.
